scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -279,9 +279,121 @@
   (-> (d/entity db [:instance/task-id task-id])
       :db/id))
 
+(histograms/defhistogram [cook-mesos scheduler progress-aggregator-pending-states])
+(meters/defmeter [cook-mesos scheduler progress-aggregator-message-rate])
+(meters/defmeter [cook-mesos scheduler progress-aggregator-drop-rate])
+(counters/defcounter [cook-mesos scheduler progress-aggregator-drop-count])
+
+(defn progress-update-aggregator
```

Comment:
My own notes:

Chan                                                                Use
progress-aggregator-chan                               Receive progress updates
progress-aggregator-trigger-chan                    "Triggers" sending a progress update on progress-state-chan
progress-state-chan                                         Sends progress-state map when progress-aggregator-trigger-chan is fired
query-state-chan                                              Receive messages (supported [query, exit]), query sends progress map on response, exit sends exited message back